core/kotlinx-coroutines-test/src/TestCoroutineDispatcher.kt

```diff
@@ -0,0 +1,296 @@
+package kotlinx.coroutines.test
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.test.internal.ThreadSafeHeap
+import kotlinx.coroutines.test.internal.ThreadSafeHeapNode
+import java.util.concurrent.TimeUnit
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Control the virtual clock time of a [CoroutineDispatcher].
+ *
+ * Testing libraries may expose this interface to tests instead of [TestCoroutineDispatcher].
+ */
+@ExperimentalCoroutinesApi
+interface DelayController {
+    /**
+     * Returns the current virtual clock-time as it is known to this Dispatcher.
+     *
+     * @param unit The [TimeUnit] in which the clock-time must be returned.
+     * @return The virtual clock-time
+     */
+    @ExperimentalCoroutinesApi
+    fun currentTime(unit: TimeUnit = TimeUnit.MILLISECONDS): Long
+
+    /**
+     * Moves the Dispatcher's virtual clock forward by a specified amount of time.
+     *
+     * The amount the clock is progressed may be larger than the requested delayTime if the code under test uses
+     * blocking coroutines.
+     *
+     * @param delayTime The amount of time to move the CoroutineContext's clock forward.
+     * @param unit The [TimeUnit] in which [delayTime] and the return value is expressed.
+     * @return The amount of delay-time that this Dispatcher's clock has been forwarded.
+     */
+    @ExperimentalCoroutinesApi
+    fun advanceTimeBy(delayTime: Long, unit: TimeUnit = TimeUnit.MILLISECONDS): Long
+
+    /**
+     * Moves the current virtual clock forward just far enough so the next delay will return.
+     *
+     * @return the amount of delay-time that this Dispatcher's clock has been forwarded.
+     */
+    @ExperimentalCoroutinesApi
+    fun advanceTimeToNextDelayed(): Long
+
+    /**
+     * Immediately execute all pending tasks and advance the virtual clock-time to the last delay.
```

Comment:
what is the use case to use `advanceTimeToNextDelay` when `advanceUntilIdle` already exists?
Unless there is a use case in mind, it might make sense to keep the API simpler and not provide `advanceTimeToNextDelay` ?

Reply:
Agreed. Will pull this since `advanceTimeBy` will allow the developer to accomplish this with more precision.